osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -112,27 +123,51 @@ private void onSearchStarted()
 
         private void onSearchFinished(List<BeatmapSetInfo> beatmaps)
         {
+            //No matches case
             if (!beatmaps.Any())
             {
-                LoadComponentAsync(new NotFoundDrawable(), addContentToPlaceholder, (cancellationToken = new CancellationTokenSource()).Token);
+                LoadComponentAsync(notFoundContent, addContentToPlaceholder, (cancellationToken = new CancellationTokenSource()).Token);
                 return;
             }
 
-            var newPanels = new FillFlowContainer<BeatmapPanel>
+            //New query case
+            if (!shouldAddNextPage)
             {
-                RelativeSizeAxes = Axes.X,
-                AutoSizeAxes = Axes.Y,
-                Spacing = new Vector2(10),
-                Alpha = 0,
-                Margin = new MarginPadding { Vertical = 15 },
-                ChildrenEnumerable = beatmaps.Select<BeatmapSetInfo, BeatmapPanel>(b => new GridBeatmapPanel(b)
+                //Spawn new child
+                var newPanels = new FillFlowContainer<BeatmapPanel>
```

Comment:
not actually "new" panels sinc eyou're appending to existing list. doesn't work.